wpilibc/src/main/native/include/frc/smartdashboard/MechanismObject2d.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include <map>
+#include <memory>
+
+#include <wpi/StringMap.h>
+
+#include <networktables/NetworkTable.h>
+
+namespace frc {
+
+class MechanismObject2d {
+  friend class Mechanism2d;
+ protected:
+  explicit MechanismObject2d(const wpi::Twine& name);
+  /**
+   * Update all entries with new ones from a new table.
+   *
+   * @param table the new table.
+   */
+  virtual void UpdateEntries(std::shared_ptr<NetworkTable> table) = 0;
+ public:
+  virtual ~MechanismObject2d() = default;
+  /**
+   * Retrieve the object's name.
+   * 
+   * @return the object's name relative to its parent.
+   */
+  const std::string& GetName() const;
+
+  /**
+   * Append a Mechanism object that is based on this one.
+   *
+   * @param name the name of the new object.
+   * @param args constructor arguments of the object type.
+   * @return the constructed and appended object, useful for variable assignments and call chaining.
+   * @throw if an object with the given name already exists.
+   */
+  template<typename T, typename... Args, typename = std::enable_if_t<std::is_convertible_v<T*, MechanismObject2d*>>>
+  T* Append(wpi::StringRef name, Args&&... args) {    
+    if (m_objects.count(name)) {
+      throw std::runtime_error(("MechanismObject names must be unique! `" + name
+                               + "` was inserted twice!").str());
+    }
+    std::unique_ptr<T> ptr = std::make_unique<T>(name, std::forward<Args>(args)...);
+    T* ex = ptr.get();
+    m_objects.try_emplace<std::unique_ptr<MechanismObject2d>>(name, std::move(ptr));

```

Comment:
You can avoid the double-lookup here by simply using [] access and checking for nullptr.  E.g.
```
auto& obj = m_objects[name];
if (obj) {
  throw
}
obj = std::make_unique<...>
```

Reply:
I see a single lookup: `count()`. Where do you see the other?

Reply:
try_emplace is a lookup.

Reply:
but we still need to insert it anyway, so what does it change? unless `obj = make_...` inserts it.

Reply:
Yes it does insert it.

Reply:
Technically, the [] access inserts it (with a default-constructed, thus nullptr value).